libs/librepcbcommon/fileio/directorylock.cpp

```diff
@@ -34,114 +35,85 @@ namespace librepcb {
  ****************************************************************************************/
 
 DirectoryLock::DirectoryLock() noexcept :
-    mFileToLock(), mLockFilepath(), mLockedByThisObject(false)
+    mDirToLock(), mLockFilePath(), mLockedByThisObject(false)
 {
     // nothing to do...
 }
 
-DirectoryLock::DirectoryLock(const FilePath& filepath) noexcept :
-    mFileToLock(), mLockFilepath(), mLockedByThisObject(false)
+DirectoryLock::DirectoryLock(const FilePath& dir) noexcept :
+    mDirToLock(), mLockFilePath(), mLockedByThisObject(false)
 {
-    setFileToLock(filepath);
+    setDirToLock(dir);
 }
 
 DirectoryLock::~DirectoryLock() noexcept
 {
-    if (mLockedByThisObject)
-        try{unlock();} catch (...) {}
+    if (mLockedByThisObject) {
+        try {
+            unlock(); // can throw
+        } catch (const Exception& e) {
+            qCritical() << "Could not remove lock file:" << e.getUserMsg();
+        }
```

Comment:
There aren't any other exceptions, all exceptions used in LibrePCB must be derived from `Exception`. If a 3rdparty library would throw other exceptions, we would need to wrap them.